hints fix

balejk opened the pull request:
- **extension: store web page for dom_element and dom_document**
- **extension: rewrite most of dom_element.__index using JSC**
- **doc: fix the type of dom_element.value**
- **extension: rewrite dom_element.__newindex using JSC**
- **extension: rewrite luaH_dom_element_style_index using JSC**
- **extension: rewrite dom_element attributes table using JSC**
- **extension: rewrite dom_element:client_rects using JSC**
- **extension: implement __newindex for dom_element.style**
- **lib: change the way hints are styled**

---

I've been sitting on this for almost a year now so it is time to put it up. I have been hoping to get some feedback on this from @c0dev0id (to whom I have mailed the patches) but unfortunately I haven't heard from them for a long time now. Most importantly I was hoping to get some help with a particular issue: when in the follow mode and inputting part of the label, the relevant targets get highlighted by the change in the overlay color, this color however doesn't reset back to the non-highlight state when the input is changed (i. e. if there are labels such as `se` and `ra`, pressing `s` highlights the former, pressing backspace and `r` highlights the latter but the former remains highlighted too). I haven't looked into this again now but it was not obvious to me back then why this happens (although it is likely trivial), so I will welcome any help.

The hints fix is implemented in the last two patches, the preceding ones are not strictly necessary but I originally based the work on them in order not to introduce more deprecated code and also because I expected they would be soon applied.

This is marked as draft because I haven't reviewed this now (but I have revisited it at least once since it was originally written and have been daily driving it since it was written with no collateral issues observed) and also because of the issue mentioned above. Also, I recall there was some issue with the documentation where I was having troubles figuring out how to reference variables/modules with non-unique names.

balejk committed:
extension: store web page for dom_element and dom_document

This will allow for a gradual transition from the deprecated WebKitDOM
API.

balejk committed:
extension: rewrite most of dom_element.__index using JSC

As a side effect, this causes the frequent

(WebKitWebProcess:xxxx): GLib-GObject-CRITICAL **: xx:xx:xx.xxx: g_object_get_is_valid_property: object class 'WebKitDOMHTML*Element' has no property named 'type'

errors to no longer appear.

balejk committed:
doc: fix the type of dom_element.value

balejk committed:
extension: rewrite dom_element.__newindex using JSC

balejk committed:
extension: rewrite luaH_dom_element_style_index using JSC

balejk committed:
extension: rewrite dom_element attributes table using JSC

balejk committed:
extension: rewrite dom_element:client_rects using JSC

balejk committed:
extension: implement __newindex for dom_element.style

balejk committed:
extension: implement __newindex for dom_element.style

balejk committed:
lib: change the way hints are styled

Inline styles or those inserted via the <style> element do not work at
pages that restrict styles with Content-Security-Policy headers. Setting
the styles on the style property (rather than attribute) of the
HTMLElement, however, does work.

Thank you to vimb for this solution.

Fixes: https://github.com/luakit/luakit/issues/708
Fixes: https://github.com/luakit/luakit/issues/510
Fixes: https://github.com/luakit/luakit/issues/760

fictitiousexistence commented:
This fixes follow not workig on miniflux for me https://github.com/luakit/luakit/issues/708

balejk committed:
lib: change the way follow hints are styled

Styles applied using the <style> element as well as those set inline
with "style" as an attribute do not work on sites with strict content
security policy. Only styles set inline with "style" as a property take
effect on such sites.

Thank you to vimb for this solution.

Fixes: https://github.com/luakit/luakit/issues/708
Fixes: https://github.com/luakit/luakit/issues/510
Fixes: https://github.com/luakit/luakit/issues/760
